turbogears/memory_profiler_setup.py

```
@@ -45,30 +53,51 @@
+    'OFF' turns off memory profiling, 'ECHO' publishes current value of the variable to the log
     :param _thread_log: console logger
     :return: 
     """
     _thread_log.info('started toggle_memory_profile_via_fifo thread PID({})'.format(os.getpid()))
-    fifo_name = TURBOGEARS_PROFILER_FIFO_PATH
+    fifo_path = TURBOGEARS_PROFILER_FIFO_PATH
+    if not os.path.exists(TURBOGEARS_PROFILER_FIFO_PATH):
+        os.mkdir(fifo_path)
+    fifo_name = os.path.join(TURBOGEARS_PROFILER_FIFO_PATH, TURBOGEARS_PROFILER_FIFO_NAME.format(os.getpid()))
     if not os.path.exists(fifo_name):
+        _thread_log.info('[{}] creating FIFO {}'.format(os.getpid(), fifo_name))
         os.mkfifo(fifo_name)
     while True:
         with open(fifo_name, 'r') as config_fifo:
-            _thread_log.info('opened FIFO toggle_memory_profile_via_fifo thread')
+            _thread_log.info('opened FIFO {} in toggle_memory_profile_via_fifo thread'.format(fifo_name))
             line = config_fifo.readline()[:-1]
-            _thread_log.info('READ LINE toggle_memory_profile_via_fifo thread ====>{}'.format(line))
-            toggle_memory_profile_logging(_thread_log)
+            state=_get_state_from_pipe_command(line)
+            _thread_log.info('READ LINE toggle_memory_profile_via_fifo thread ====>{}, state: {}'.format(line, state.name))
+            if state == MemoryProfilerState.ON or state == MemoryProfilerState.OFF:
+                set_memory_profile_logging(_thread_log, state.value[2])
+            elif state == MemoryProfilerState.ECHO:
+                _thread_log.info('ECHO MEMORY_PROFILE_LOGGING_ON ==> {}'.format(get_memory_profile_logging_on()))
+
+
+def _get_state_from_pipe_command(command):
+    if command.lower() not in [MemoryProfilerState.ON.value[1],
+                               MemoryProfilerState.OFF.value[1],
+                               MemoryProfilerState.ECHO.value[1]]:
+        return MemoryProfilerState.UNKNOWN
```

We sure we want to fail silently here?